c2corg_ui/static/js/documentediting.js

```diff
@@ -354,5 +382,204 @@ app.DocumentEditingController.prototype.getCoordinatesFromPoint_ = function(
 };
 
 
+/**
+ * Clone input
+ * @export
+ */
+
+app.DocumentEditingController.prototype.addAnotherMapsInfo = function() {
+  var template = $('#maps_info-group input:last-of-type').val('').clone();
+  template.insertBefore($('#document-add-maps'));
+}
+
+
+/**
+ * Navigate through creation/editing steps
+ * @param {number} step
+ * @param {string} direction
+ * @export
+ */
+
+app.DocumentEditingController.prototype.step = function(step, direction) {
+  if (step === 1) {
+    $('.editing').animate({left: '0'});
+    this.animateBar_(step, direction);
+    this.previous_step = 0;
+    this.next_step = 2;
+  }
+  if (step === 2) {
+    $('.editing').animate({left: '-115%'});
+    this.animateBar_(step, direction);
+    this.previous_step = 1;
+    this.next_step = 3;
+  }
+  if (step === 3) {
+    $('.editing').animate({left: '-229%'});
+    this.animateBar_(step, direction);
+    this.previous_step = 2;
+    this.next_step = 4;
+  }
+  if (step === 4) {
+    $('.editing').animate({left: '-343%'});
+    this.animateBar_(step, direction);
+    this.previous_step = 3;
+    this.next_step = 5;
+  }
+  if (step === 5) {
+    $('.editing').animate({left: '-457%'});
+    this.animateBar_(step, direction);
+    this.previous_step = 4;
+    this.next_step = 6;
+  }
+}
+
+
+/**
+ * Animate the progress bar
+ * @param {number} step
+ * @param {string} direction
+ * @private
+ */
+
+app.DocumentEditingController.prototype.animateBar_ = function(step, direction) {
+  var percent = 100 / this.max_steps;
+  var that = this;
```

Comment:
You can avoid the `that` by using `bind`:

     var timeout = setInterval(function() {...}.bind(this));